### PR TITLE
Added support for redis gears trigger function commands

### PR DESCRIPTION
**Pull request by vladvildanov**

### CODE REVIEW

File: .github/workflows/stack.yml [hunk 1]
```diff
@@ -21,6 +21,11 @@ jobs:
         ports:
           - 6379:6379
 
+      redis-gears:
```
[the remark this answers is not in the document]
Comment (vladvildanov) [reply .github/workflows/stack.yml:24]: Oh, didn't knew about that, sure we can

File: src/Command/Redis/TFCALLASYNC.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * (c) 2009-2020 Daniele Alessandri
+ * (c) 2021-2023 Till Krüss
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Predis\Command\Redis;
+
+class TFCALLASYNC extends TFCALL
```
Comment (chayim) [inline src/Command/Redis/TFCALLASYNC.php:15]: @shacharpash this is what I meant with [this comment](https://github.com/redis/redis-py/pull/2861#discussion_r1280260752) in redis-py. I think you two should align on an API

**Review by chayim — APPROVED** [review]
Approved pending the conversation.

File: tests/Predis/Command/Redis/BloomFilter/BFADD_Test.php [hunk 3]
```diff
@@ -61,6 +61,7 @@ public function testParseResponse(): void
 
     /**
      * @group connected
+     * @group relay-incompatible
```
Comment (tillkruss) [inline tests/Predis/Command/Redis/BloomFilter/BFADD_Test.php:64]: Why all these `@group relay-incompatible`? They should be able to be called using `rawCommand()`.

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: tests/Predis/Command/Redis/BloomFilter/BFADD_Test.php (hunk 3 above)
Comment (tillkruss) [reply tests/Predis/Command/Redis/BloomFilter/BFADD_Test.php:64]: @vladvildanov: How about we call all the newly added groups `relay-resp3` and ignore it too. That will make it easier to "undo" this change without having to sort through the groups.

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: tests/Predis/Command/Redis/BloomFilter/BFADD_Test.php (hunk 3 above)
Comment (vladvildanov) [reply tests/Predis/Command/Redis/BloomFilter/BFADD_Test.php:64]: @tillkruss Agreed

Comment (vladvildanov) [reply tests/Predis/Command/Redis/BloomFilter/BFADD_Test.php]: @tillkruss Fixed

**Review by tillkruss — CHANGES_REQUESTED** [review]
Can we add CHANGELOG entries as part of this PR?

**Comment by vladvildanov** [comment]
@tillkruss Changelogs updated as separate PR for whole 3.0 update

https://github.com/predis/predis/pull/1393

**Comment by vladvildanov** [comment]
@tillkruss Needs update on this, it's ready to merge

**Review by chayim — APPROVED** [review]

File: src/Connection/Cluster/ClusterInterface.php [hunk 4]
```diff
@@ -20,4 +21,11 @@
  */
 interface ClusterInterface extends AggregateConnectionInterface
 {
+    /**
+     * Executes given command on each connection from connection pool.
+     *
+     * @param  CommandInterface $command
+     * @return array
+     */
+    public function executeCommandOnEachNode(CommandInterface $command): array;
```
Comment (tillkruss) [inline src/Connection/Cluster/ClusterInterface.php:30]: This PR contains substantial changes. Please document everything in the `CHANGELOG.md`.

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: src/Connection/Cluster/ClusterInterface.php (hunk 4 above)
Comment (vladvildanov) [reply src/Connection/Cluster/ClusterInterface.php:30]: @tillkruss Done

File: src/Connection/Replication/SentinelReplication.php [hunk 5]
```diff
@@ -315,31 +316,30 @@ public function getSentinelConnection()
...
-                        'role' => 'sentinel',
-                    ];
-                }
-            } catch (ConnectionException $exception) {
-                $this->sentinelConnection = null;
-
-                goto SENTINEL_QUERY;
+        try {
+            $payload = $sentinel->executeCommand(
+                RawCommand::create('SENTINEL', 'sentinels', $this->service)
```
Comment (tillkruss) [inline src/Connection/Replication/SentinelReplication.php:324]: Let's remove this code changes please.

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: src/Connection/Replication/SentinelReplication.php (hunk 5 above)
Comment (vladvildanov) [reply src/Connection/Replication/SentinelReplication.php:324]: @tillkruss This changes related to phpcsfixer library update that used as a linter and step in CI pipeline, this changes how nothing code wise it's all about indents. If I remove this changes CI will fail

Comment (tillkruss) [reply src/Connection/Replication/SentinelReplication.php:324]: That was disabled: https://github.com/predis/predis/commit/a9d18b182ca1efe43ad3066520fe1bf566b5c189

Comment (vladvildanov) [reply src/Connection/Replication/SentinelReplication.php]: Ah, perfect! Changes was reverted

**Review by tillkruss — APPROVED** [review]

**Pull request merged**